src/TaskViewSection/Task.tsx

```diff
@@ -0,0 +1,107 @@
+/* eslint-disable react/no-array-index-key */
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import { ChangeEvent, useContext } from 'react'
+import { v4 as uuid } from 'uuid'
+import TaskModal from './TaskModal/TaskModal'
+import { colorStyleCheckboxHandler } from '../Helpers/colorStyleClassHandler'
+import TaskViewSectionContext from '../Contexts/TaskViewSectionContext'
+import { TaskType } from '../Contexts/TaskContext'
+
+type TaskProps = {
+    task: TaskType
+    updateTaskStatus: (e: ChangeEvent<HTMLInputElement>) => void
+    searchInput: string
+}
+
+const Task = ({ task, updateTaskStatus, searchInput }: TaskProps) => {
+    const { selectedCategoryTab } = useContext(TaskViewSectionContext)
+    const correctCategory =
+        selectedCategoryTab === 'all' || task.category === selectedCategoryTab
+    const correctName =
+        !searchInput ||
+        task.name.toLowerCase().includes(searchInput.toLowerCase())
```

Comment:
question: why do we check this `task.name.toLowerCase().includes(searchInput.toLowerCase())` in the end?